### PR TITLE
Fix comment ignore with trailing comment not recognized

### PR DESCRIPTION
**Pull request by oliversun9**
This fixes the problem where comment ignores with a trailing comment is not recognized as a comment ignore:

```
// buf:lint:ignore PACKAGE_DIRECTORY_MATCH some comment ...
```

### CODE REVIEW

File: private/bufpkg/bufcheck/client.go [hunk 1]
```diff
@@ -491,27 +491,25 @@ func ignoreLocation(
...
 // checkCommentLineForCheckIgnore checks that the comment line starts with the configured
-// comment ignore prefix, and the rest of the string is the ruleID of the check.
+// comment ignore prefix, a space and the ruleID of the check.
 //
-// We currently do not support multiple rules per comment ignore:
+// All of the following comments are valid, ignoring SERVICE_PASCAL_CASE and this rule only:
 //
-//	Invalid:
-//		// buf:lint:ignore SERVICE_SUFFIX, SERVICE_PASCAL_CASE
+//	// buf:lint:ignore SERVICE_PASCAL_CASE, SERVICE_SUFFIX
```
Comment (bufdev) [inline private/bufpkg/bufcheck/client.go:498]: Is this saying that you can do comma-separated ignores? I don't think that's in scope

Comment (oliversun9) [reply private/bufpkg/bufcheck/client.go]: Updated comment in [f516a9c](https://github.com/bufbuild/buf/pull/3295/commits/f516a9c597e110b619aed297cde8d96e723b3f3a) (it also said `ignoring SERVICE_PASCAL_CASE and this rule only` on top)

**Review by bufdev — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,7 +2,8 @@
 
 ## [Unreleased]
 
-- No changes yet.
+- Fix issue with `buf lint` where comment ignores in the shape of `// buf:lint:ignore <RULE_ID> <extra comment>`
+  are not recognized due to the extra comment.
```
Comment (bufdev) [inline CHANGELOG.md:6]: `were not recognized` not `are not recognized`

Comment (oliversun9) [reply CHANGELOG.md]: fixed in [78c78e6](https://github.com/bufbuild/buf/pull/3295/commits/78c78e6e49151ff5d133f49ab14533deae75336b)

**Review by bufdev — APPROVED** [review]

**Pull request merged**